### PR TITLE
add async test client (GSI-99)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
Test client for ASGI apps that can be used in context of a running event loop.

Bumps version to 0.3.0.
```

### CODE REVIEW

File: ghga_service_commons/api/testing.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @app.get("/")
+    def index():
+        return "Hello World"
+
+    @pytest.mark.asyncio
+    async def test_index():
+        async with AsyncTestClient(app) as client:
+            response = await client.get("/")
+
+    assert response.status == 200
```
Comment (Cito) [inline ghga_service_commons/api/testing.py:51]:
```suggestion
    assert response.status_code == 200
```

File: ghga_service_commons/api/testing.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    assert response.content == "Hello World"
```
Comment (Cito) [inline ghga_service_commons/api/testing.py:52]:
```suggestion
    assert response.json() == "Hello World"
```

Alternatively,
```py
assert response.content== b'"Hello World"'
assert response.text == '"Hello World"'
```

File: ghga_service_commons/api/testing.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+    """
+
+    def __init__(self, app: Callable[..., Any]):
+        """Initialize with ASGI app."""
+
+        port = get_free_port()
```
[the remark this answers is not in the document]
Comment (Cito) [reply ghga_service_commons/api/testing.py:59]: But then this AsyncTestClient does not really bring and added value over the httpx.AsyncClient.

**Review by Cito — CHANGES_REQUESTED** [review]

File: ghga_service_commons/api/testing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply ghga_service_commons/api/testing.py:59]: I thought so, too, but without the port, it did not work for both the upload controller service and metldata. Maybe we should investigate that at some point. Should I put this onto the refactoring list or do you have an immediate idea?

[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply ghga_service_commons/api/testing.py]: Yes, exactly. It is this test that is failing if you don't specify a base_url: https://github.com/ghga-de/metldata/blob/feature/artifacts_rest_GSI-30/tests/artifact_rest/test_api_factory.py#L97

I could confirm, however, that it is not essential to pick a free port. I changed the PR accordingly. However, the ability to pick a free port is still relevant and is needed elsewhere for the implementation of testcontainers. So keeping it here might still be worth it.

Thanks for pointing that out.

**Review by Cito — APPROVED** [review]

**Pull request merged**